checkmarx-ast-eclipse-plugin/src/com/checkmarx/eclipse/views/DataProvider.java

```diff
@@ -366,33 +366,95 @@ private List<DisplayModel> addResults(String scanId, Map<String, List<DisplayMod
 		return returnList;
 	}
 
+
+	/**
+	 * Creates a clean Result object with decoded HTML entities
+	 *
+	 * @param resultItem Original result object
+	 * @return New Result object with cleaned values
+	 */
+	private Result createCleanResult(Result resultItem) {
+	    String cleanDescription = resultItem.getDescription() != null ?
```

Comment:
can it be that resultItem will be null sometimes? (even if it error, we don't want to get null pointer exception

Reply:
there's no need to check for null in the "transform" method because we already have a null check before calling it.
In the "processResults" method, Line 250:

if(scanResults == null || scanResults.getResults() == null || scanResults.getResults().isEmpty()) {
    return Collections.emptyList();
}